cxx-squid/src/test/java/org/sonar/cxx/visitors/CxxCognitiveComplexityVisitorTest.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Sonar C++ Plugin (Community)
+ * Copyright (C) 2010-2017 SonarOpenCommunity
+ * http://github.com/SonarOpenCommunity/sonar-cxx
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.cxx.visitors;
+
+import static org.fest.assertions.Assertions.assertThat;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+
+import com.sonar.sslr.api.AstNode;
+import com.sonar.sslr.api.AstNodeType;
+import com.sonar.sslr.api.Grammar;
+import org.junit.Test;
+import org.sonar.cxx.api.CxxMetric;
+import org.sonar.cxx.CxxFileTester;
+import org.sonar.cxx.CxxFileTesterHelper;
+import org.sonar.cxx.CxxAstScanner;
+import org.sonar.cxx.parser.CxxGrammarImpl;
+import org.sonar.squidbridge.api.SourceFile;
+
+public class CxxCognitiveComplexityVisitorTest {
+
+  private void testFile(String fileName, int expectedCognitiveComplexity) throws UnsupportedEncodingException, IOException {
+    CxxCognitiveComplexityVisitor<Grammar> visitor = CxxCognitiveComplexityVisitor.<Grammar>builder()
+      .setMetricDef(CxxMetric.COGNITIVE_COMPLEXITY)
+      .build();
+
+    CxxFileTester tester = CxxFileTesterHelper.CreateCxxFileTester(fileName, ".");
+    SourceFile sourceFile = CxxAstScanner.scanSingleFile(tester.cxxFile, tester.sensorContext, CxxFileTesterHelper.mockCxxLanguage(), visitor);
+
+    assertThat(sourceFile.getInt(CxxMetric.COGNITIVE_COMPLEXITY)).isEqualTo(expectedCognitiveComplexity);
```

Comment:
Assert should be implemented in tests and not in helper method. This is a minor issue and could be improved in another PR e.g. private int testFile(String fileName) returns CxxMetric.COGNITIVE_COMPLEXITY

Reply:
I was just about to change this to go with another PR, but you've already fixed it, thanks!